legacy/src/main/java/com/amazon/opendistroforelasticsearch/sql/legacy/plugin/RestSQLQueryAction.java

```diff
@@ -179,4 +199,33 @@ private void sendResponse(RestChannel channel, RestStatus status, String content
         status, "application/json; charset=UTF-8", content));
   }
 
+  private void reportError(RestChannel channel, Exception e, RestStatus status) {
+    sendResponse(
+        channel, status, ErrorMessageFactory.createErrorMessage(e, status.getStatus()).toString());
+  }
+
+  private static void logAndPublishMetrics(Exception e) {
+    if (isClientError(e)) {
+      LOG.error(LogUtils.getRequestId() + " Client side error during query execution", e);
+      Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_CUS).increment();
+    } else {
+      LOG.error(LogUtils.getRequestId() + " Server side error during query execution", e);
+      Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_SYS).increment();
+    }
+  }
+
+  private static boolean isClientError(Exception e) {
+    return e instanceof NullPointerException // NPE is hard to differentiate but more likely caused by bad query
+        || e instanceof SqlParseException
+        || e instanceof ParserException
+        || e instanceof SQLFeatureNotSupportedException
+        || e instanceof SQLFeatureDisabledException
+        || e instanceof IllegalArgumentException
+        || e instanceof IndexNotFoundException
+        || e instanceof VerificationException
+        || e instanceof SqlAnalysisException
+        || e instanceof SyntaxCheckException
+        || e instanceof SemanticCheckException;
```

Comment:
Because this is in new engine, exceptions like `SqlParseException`, `ParserException` and some others won't happen. Probably only `SyntaxCheckException` and `QueryEngineException` are required same as here: https://github.com/opendistro-for-elasticsearch/sql/blob/85202b67492fbc2924ec40b90cad9c5475048708/protocol/src/main/java/com/amazon/opendistroforelasticsearch/sql/protocol/response/format/JdbcResponseFormatter.java#L104

Reply:
Another thought: since earlier we've already passed "compiling process" in new engine, all exception occurs in execution engine should be treated as server error?